ClientCore/CnCNet5/CnCNetGame.cs

```diff
@@ -27,13 +27,22 @@ public class CnCNetGame
         /// </summary>
         public string GameBroadcastChannel { get; set; }
 
+
+        /// <summary>
+        /// The IRC game pattern of a hosted game channels
+        /// </summary>
+        public string GameListPattern { get; set; }
+
+
         /// <summary>
         /// The executable name of the game's client.
         /// </summary>
         public string ClientExecutableName { get; set; }
 
         public Texture2D Texture { get; set; }
 
+        public Texture2D TextureSpecialGameMode { get; set; }
+
```

Comment:
Same, the solution is hacky, which in this case also goes against the design (this is a distribution/game/mod icon, it can't change depending on a game mode). We would need to come up with a proper way to achieve what you want.

An option I see is:
0. Use both RA2/YR icon as `Texture` for CnCNet YR,
1. Make an option in ClientDefintions to hide the main game icon in game list,
2. Make an option to add enabled/disabled icons to the game settings, something like `EnabledIcon` and `DisabledIcon` for INI defined game options, and `Icon` for dropdown items,
3. When some option is broadcasted to the outside world - prepend an icon to the game name, same for all of the options.

This way you could define ra2mode as broadcasted to cncnet lobby in INI, define enabled / disabled texture for it and it would change the "icon" of the game.